python/rmm/_lib/device_buffer.pyx

```diff
@@ -72,12 +72,13 @@ cdef class DeviceBuffer:
         if s == 0:
             return b""
 
+        cdef cudaStream_t c_stream = <cudaStream_t>stream
```

Comment:
IIRC this needs a double cast to handle using integers as the opaque handle for a stream in Python land.
```suggestion
        cdef cudaStream_t c_stream = <cudaStream_t><uintptr_t>stream
```

Reply:
In this case `stream` is already typed as `uintptr_t`. So we should be good here 🙂

Reply:
Github UI hid this from me 😄, looks good